Add `editor::BlameHover` action for triggering the blame popover via keyboard

Daste745 opened the pull request:
Make the git blame popover available via the keymap by making it an action. The blame popover stays open after being shown via the action, similar to the `editor::Hover` action.

I added a default vim-mode key binding for `g b`, which goes in hand with `g h` for hover. I'm not sure what the keybind would be for regular layouts, if any would be set by default.

I'm opening this as a draft because I coludn't figure out a way to position the popover correctly above/under the cursor head. I saw some uses of `content_origin` in other places for calculating absolute pixel positions, but I'm not sure how to make use of it here without doing a big refactor of the blame popover code 🤔. I would appreciate some help/tips with positioning, because it seems like the last thing to implement here.

Opening as a draft for now because I think without the correct positioning this feature is not complete.

Closes https://github.com/zed-industries/zed/discussions/26447

Release Notes:

- Added `editor::BlameHover` action for showing the git blame popover under the cursor. By default bound to `ctrl-k ctrl-b` and to `g h` in vim mode.

Daste745 committed:
editor: Add bindable `blame_hover` action

Daste745 committed:
editor: Bind BlameHover to `g b` in vim mode

Daste745 committed:
editor: Show blame popover on the `editor::BlameHover` action

Daste745 committed:
editor: Ignore delay when showing blame popover with keyboard

Daste745 committed:
editor: WIP Show blame popover on the cursor's position

Daste745 committed:
editor: Don't hide blame popover when triggered via keyboard action

ConradIrwin commented:
We already have `K` (or `gh`) for the other hovers in vim mode, so would be nice to not need ctrl.

Daste745 committed:
editor: Change BlameHover keymap to `g B` in vim mode

Daste745 committed:
editor: Bind BlameHover to `ctrl-k ctrl-b`

Daste745 committed:
editor: Offset keyboard-initiated blame popovers by the editor's origin

Daste745 commented:
Rebased all my changes onto main, implemented the positioning fix and changed the default keymap.

I ended up with the following default keymap:
- linux: `ctrl-k ctrl-b`
- macos: `cmd-k cmd-b`
- vim: `g B`

Adding `self.last_bounds.origin` to the newest anchor's position ended up fixing the relative positioning issue. The popover is always above/under the latest cursor and doesn't cover up the line.

Please let me know if we need any further changes to complete this.

ConradIrwin commented:
Nice, thanks for this!
